scripts/reboot

```diff
@@ -242,9 +248,19 @@ if [ -x ${DEVPATH}/${PLATFORM}/${PLATFORM_UPDATE_REBOOT_CAUSE} ]; then
     ${DEVPATH}/${PLATFORM}/${PLATFORM_UPDATE_REBOOT_CAUSE}
 fi
 
+if [ -x ${DEVPATH}/${PLATFORM}/${PRE_REBOOT_HOOK} ]; then
+    debug "Executing the pre-reboot script"
+    ${DEVPATH}/${PLATFORM}/${PRE_REBOOT_HOOK}
+fi
```

Comment:
If this script fails we might want to abort the reboot operation (unless it is beeing forced).
Also, in case it fails, I wonder whether we should update reboot cause by above ```if``` block?